src/main/java/org/jvnet/hudson/update_center/util/UrlToGitHubSlugConverter.java

```diff
@@ -0,0 +1,30 @@
+package org.jvnet.hudson.update_center.util;
+
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.StringUtils;
+
+public final class UrlToGitHubSlugConverter {
+
+    private UrlToGitHubSlugConverter() {
+    }
+
+    /**
+     * Converts a github url to its 'slug'
+     * @param url a url like: https://github.com/jenkinsci/blueocean-plugin
+     * @return the github slug, jenkinsci/blueocean-plugin
+     */
+    public static String convert(@Nonnull String url) {
```

Comment:
Do we expect strings like `http://github.com/org/repo`, `https:/github.com/org/repo`, `https://github.com/org/repo/repo`, `git@github.com:org/repo.git`, etc. to get here?

Reply:
From what I remember it’s using the already resolved repo name from earlier but will check when I look at adapting this to graphql